eng/common/scripts/get-markdown-files-from-changed-files.ps1

```diff
@@ -0,0 +1,47 @@
+param (
+    # The root repo we scaned with.
+    [string[]] $RootRepo = "./",
+    # The target branch to compare with.
+    [string] $targetBranch = "origin/${env:SYSTEM_PULLREQUEST_TARGETBRANCH}"
+)
+$deletedFiles = (git diff origin/$targetBranch HEAD --name-only --diff-filter=D)
+$renamedFiles = (git diff origin/$targetBranch HEAD --diff-filter=R)
+$changedMarkdowns = (git diff origin/$targetBranch HEAD --name-only -- '*.md')
+
+$beforeRenameFiles = @()
+# Retrieve the renamed from files.
+foreach($file in $renamedFiles) {
+    if ($file -match "^rename from (.*)$") {
+        $beforeRenameFiles += $file -replace "^rename from (.*)$", '$1'
+    }
+}
+# A combined list of deleted and renamed files.
+$relativePathLinks = ($deletedFiles + $beforeRenameFiles)
+# Removed the deleted markdowns. 
+$changedMarkdowns = $changedMarkdowns | Where-Object { $deletedFiles -notcontains $_ }
+# Scan all markdowns and find if it contains the deleted or renamed files.
+$markdownContainLinks = @()
+$allMarkdownFiles = Get-ChildItem -Path $RootRepo -Recurse -Include *.md
+foreach ($f in $allMarkdownFiles) {
+    $filePath = $f.ToString()
+    $content = Get-Content -Path $filePath -Raw
+    foreach($l in $relativePathLinks) {
+        if ($content -match $l) {
+            $markdownContainLinks += $filePath
+            break
+        }
+    }
+}
+
+# Convert markdowns path of the PR to absolute path.
+$adjustedReadmes = $changedMarkdowns | Foreach-Object {Resolve-Path $_}
+$markdownContainLinks += $adjustedReadmes
+
+# Get rid of any duplicated ones.
+$allMarkdowns = [string[]]($markdownContainLinks | Sort-Object | Get-Unique)
```

Comment:
So are there cases here where we return null or empty string? I'm mostly asking because of the other change to Verify-Links.ps1.

Reply:
Yes, it could be empty string array if no renamed or deleted files and no readme changes.

Reply:
```
if ($urls) {
  if ($urls.Count -eq 0) {
    Write-Host "Usage $($MyInvocation.MyCommand.Name) <urls>";
    exit 1;
  }  
}
```
We have handling for the empty string array

Reply:
We will print something "Found 0 links in page" once the urls is empty. I will double check before merge in.